pkg/pool-weighted/test/InvestmentPool.test.ts

```diff
@@ -124,4 +74,214 @@ describe('InvestmentPool', function () {
+            await expect(pool.setSwapEnabled(other, false)).to.be.revertedWith('SENDER_NOT_ALLOWED');
+          });
+        });
+
+        context('when the sender is the owner', () => {
+          sharedBeforeEach('set sender to owner', async () => {
+            sender = owner;
+            await pool.init({ from: sender, initialBalances });
+          });
+
+          it('swaps can be enabled and disabled', async () => {
+            await pool.setSwapEnabled(sender, false);
+            expect(await pool.instance.getSwapEnabled()).to.be.false;
+
+            await pool.setSwapEnabled(sender, true);
+            expect(await pool.instance.getSwapEnabled()).to.be.true;
+          });
+
+          it('disabling swaps emits an event', async () => {
+            const receipt = await pool.setSwapEnabled(sender, false);
+
+            expectEvent.inReceipt(await receipt.wait(), 'SwapEnabledSet', {
+              swapEnabled: false,
+            });
+          });
+
+          it('enabling swaps emits an event', async () => {
+            const receipt = await pool.setSwapEnabled(sender, true);
+
+            expectEvent.inReceipt(await receipt.wait(), 'SwapEnabledSet', {
+              swapEnabled: true,
+            });
+          });
+
+          it('owner can join and receive BPT, then exit', async () => {
+            const bptBeforeJoin = await pool.balanceOf(owner.address);
+            await expect(pool.joinGivenIn({ from: owner, amountsIn: initialBalances })).to.not.be.reverted;
+
+            const bptAfterJoin = await pool.balanceOf(owner.address);
+            expect(bptAfterJoin).to.gt(bptBeforeJoin);
+
+            await expect(pool.exitGivenOut({ from: owner, amountsOut: initialBalances })).to.not.be.reverted;
+            const bptAfterExit = await pool.balanceOf(owner.address);
+            expect(bptAfterExit).to.lt(bptAfterJoin);
+          });
+
+          context('when swaps disabled', () => {
+            sharedBeforeEach(async () => {
+              await pool.setSwapEnabled(sender, false);
+            });
+
+            it('disallows disproportionate joins (single token)', async () => {
+              const bptOut = await pool.balanceOf(sender);
+
+              await expect(pool.joinGivenOut({ from: sender, bptOut, token: poolTokens.get(0) })).to.be.revertedWith(
+                'INVALID_JOIN_EXIT_KIND_WHILE_SWAPS_DISABLED'
+              );
+            });
+
+            it('allows proportionate joins', async () => {
+              const startingBpt = await pool.balanceOf(sender);
+
+              const { amountsIn } = await pool.joinAllGivenOut({ from: sender, bptOut: startingBpt });
+
+              const endingBpt = await pool.balanceOf(sender);
+              expect(endingBpt).to.be.gt(startingBpt);
+              expect(amountsIn).to.deep.equal(initialBalances);
+            });
+
+            it('disallows disproportionate exits (single token)', async () => {
```

Comment:
The first and third tests should not be called the same, as they are testing different join kinds. I'd consider having two describe blocks, one for proportional join kinds and one for disproportional ones, to make reading these easier.